packages/babel-parser/src/tokenizer/index.js

```diff
@@ -175,14 +174,51 @@ export default class Tokenizer extends ParserErrors {
     return this.state.type === type;
   }
 
-  // TODO
+  /**
+   * Create a LookaheadState from current parser state
+   *
+   * @param {State} state
+   * @returns {LookaheadState}
+   * @memberof Tokenizer
+   */
+  createLookaheadState(state: State): LookaheadState {
+    return {
+      pos: state.pos,
+      value: null,
+      type: state.type,
+      start: state.start,
+      end: state.end,
+      lastTokEnd: state.end,
+      context: [this.curContext()],
```

Comment:
We don't have to copy the whole context because lookahead never update contexts.

Reply:
Here you got 2x `state.end`. It's better to allocate some memory for a var and avoid duplicate code and 2x. object access on each run